tiles/src/main/java/com/protomaps/basemap/feature/CountryInfos.java

```diff
@@ -0,0 +1,277 @@
+    Nigeria|NG|6.6|11.0
+    Nicaragua|NI|7.7|11.0
+    Netherlands|NL|8.6|11.0
+    Nepal|NP|7.7|11.0
+    Nauru|NR|9.0|11.0
+    Niue|NU|18.0|18.0
+    New Zealand|NZ|8.5|11.3
+    Oman|OM|8.7|11.0
+    Panama|PA|7.7|11.0
+    Peru|PE|6.6|11.0
+    French Polynesia|PF|11.0|11.0
+    Papua New Guinea|PG|7.0|11.0
+    Philippines|PH|8.0|11.0
+    Pakistan|PK|5.0|10.5
+    Poland|PL|6.7|11.0
+    Saint Pierre and Miquelon|PM|11.0|11.0
+    Pitcairn Islands|PN|18.0|18.0
+    Puerto Rico|PR|11.0|11.0
+    Palestine|PS|18.0|18.0
+    Portugal|PT|8.0|11.0
+    Palau|PW|11.0|11.0
+    Paraguay|PY|6.7|11.0
+    Qatar|QA|8.7|11.0
+    Romania|RO|8.0|11.0
+    Serbia|RS|9.0|11.0
+    Russia|RU|5.0|10.2
+    Rwanda|RW|8.7|11.0
+    Saudi Arabia|SA|6.6|11.0
+    Solomon Islands|SB|7.0|11.0
+    Seychelles|SC|11.0|11.0
+    Sudan|SD|6.6|11.0
+    Sweden|SE|6.7|11.0
+    Singapore|SG|11.0|11.0
+    Saint Helena|SH|11.0|11.0
+    Slovenia|SI|11.0|11.0
+    Slovakia|SK|7.7|11.0
+    Sierra Leone|SL|7.8|11.0
+    San Marino|SM|11.0|11.0
+    Senegal|SN|7.7|11.0
+    Somalia|SO|7.0|11.0
+    Suriname|SR|8.0|11.0
+    South Sudan|SS|6.6|11.0
+    Sao Tome and Principe|ST|11.0|11.0
+    El Salvador|SV|10.0|11.0
+    Sint Maarten|SX|11.0|11.0
+    Syria|SY|7.7|11.5
+    eSwatini|SZ|8.7|11.0
+    Turks and Caicos Islands|TC|11.0|11.0
+    Chad|TD|6.6|11.0
+    French Southern Antarctic Lands|TF|11.0|11.0
+    Togo|TG|7.7|11.0
+    Thailand|TH|8.2|11.0
+    Tajikistan|TJ|6.7|11.0
+    Timor-Leste|TL|9.0|11.0
+    Turkmenistan|TM|6.6|11.0
+    Tunisia|TN|7.7|11.0
+    Tonga|TO|11.0|11.0
+    Turkiye|TR|7.0|11.0
+    Trinidad and Tobago|TT|10.0|11.0
+    Tuvalu|TV|18.0|18.0
+    Taiwan|TW|8.7|11.0
+    Tanzania|TZ|6.7|11.0
+    Ukraine|UA|6.7|11.0
+    Uganda|UG|10.0|11.0
+    U.S. Minor Outlying Islands|UM|11.0|11.0
+    United States of America|US|3.5|7.5
+    Uruguay|UY|8.0|11.0
+    Uzbekistan|UZ|6.6|11.0
+    Vatican|VA|18.0|18.0
+    Saint Vincent and Grenadines|VC|11.0|11.0
+    Venezuela|VE|7.1|11.3
+    British Virgin Islands|VG|18.0|18.0
+    U.S. Virgin Islands|VI|11.0|11.0
+    Vietnam|VN|8.3|11.0
+    Vanuatu|VU|9.0|11.0
+    Wallis and Futuna Islands|WF|9.0|11.0
+    Samoa|WS|9.0|11.0
+    Kosovo|XK|10.0|11.0
+    Yemen|YE|8.7|11.0
+    South Africa|ZA|4.6|10.1
+    Zambia|ZM|6.6|11.0
+    Zimbabwe|ZW|6.7|11.0
+    """;
+
+  public record CountryInfo(String name, String isoCode, double minZoom, double maxZoom) {}
+
+  private static final HashMap<String, CountryInfo> countryInfoByName;
+  static CountryInfo unknownInfo = new CountryInfo("UNKNOWN_COUNTRY", "XX", 8.0, 11.0);
+
+  static {
+    countryInfoByName = new HashMap<>();
+    Scanner s = new Scanner(data);
+    while (s.hasNextLine()) {
+      String line = s.nextLine();
+      String[] parts = line.split("\\|");
+      countryInfoByName.put(parts[0],
+        new CountryInfo(parts[0], parts[1], Double.parseDouble(parts[2]) - 1.0, Double.parseDouble(parts[3]) - 1.0));
```

Comment:
@nvkelso what was the reason for subtracting 1 here, can we just move those into the above data? and do we need to store floats if they're only used as integers for now?

Intentionally making this flexible and type-safe so we can seamlessly add more data columns later.

Reply:
Exported min zooms for Tilezen are legacy 256 px tile size zoom values, while Planetiler prefers 512 pixel tile size zooms.

Zoom = scale so should generally be constant irrespective of tile extent (which could change to 1024 px extents in the future).

Reply:
These are sometimes floats (and min zoom should always be thought of as floats with single tenths decimal precision (1.1 and 1.0)